src/hb-ot-name-table.hh

```diff
@@ -97,12 +142,39 @@ struct NameRecord
     return UNSUPPORTED;
   }
 
-  NameRecord* copy (hb_serialize_context_t *c, const void *base) const
+  NameRecord* copy (hb_serialize_context_t *c, const void *base,
+                    const hb_hashmap_t<unsigned, const char*> *name_table_overrides) const
   {
     TRACE_SERIALIZE (this);
+    auto snap = c->snapshot ();
     auto *out = c->embed (this);
     if (unlikely (!out)) return_trace (nullptr);
-    out->offset.serialize_copy (c, offset, base, 0, hb_serialize_context_t::Tail, length);
+    if (name_table_overrides->has (nameID)) {
+      const char *name_str = name_table_overrides->get (nameID);
+      unsigned len = strlen (name_str);
+
+      unsigned text_size = 4 * len + 1;
```

Comment:
Don't do this. Get `encoded_len` by calling `hb_ot_name_convert_utf` one time first with nullptr as text_size and output.